static/js/components/menu.js

```diff
@@ -0,0 +1,115 @@
+import $ from 'jquery';
```

Comment:
LOTS changed here. Since the Craft starter kit is already using jQuery, I rewrote a bunch of this to use that instead of the custom `on` and `stateMachine` stuff.